code/modules/shuttle/supply.dm

```diff
@@ -596,3 +589,135 @@ GLOBAL_LIST_EMPTY(exports_types)
 		temp = null
 
 	updateUsrDialog()
+
+//A tablet that can order requisition goodies; when it's linked to a beacon it will deliver the goods via launch pods nearby the beacon
+
+/obj/item/req_tablet
+	name = "requisition tablet"
+	desc = "A handheld tablet that allows the average user to remote order supply packages. When this is linked to a beacon, it will deliver the goods VIA launch pods. \
+			The pods will drop in a 5x5 square around the beacon and will not injure or explode, it will teleport away shortly after dropping off the contents."
+	var/obj/item/radio/beacon/linked_beacon
+	icon = 'icons/obj/items/req_tablet.dmi'
+	icon_state = "req_tablet_off"
+	var/podType = /obj/structure/closet/supplypod/centcompod //No explosion, fast landing and teleports itself out of the landing destination
+	var/temp //TGUI things
+	var/last_viewed_group = "categories"
+
+/obj/item/req_tablet/examine(mob/user)
+	. = ..()
+	if(!linked_beacon)
+		to_chat(user, "There seems to be no beacon attached to this tablet, hit a beacon with a tablet to link to it.")
+	else
+		to_chat(user, "There seems to be a beacon attached to this tablet. Should be good to go for delivery.")
+	return .
+
+/obj/item/req_tablet/afterattack(atom/A, mob/user, proximity)
+	if(istype(A, /obj/item/radio/beacon))
+		to_chat(user, "You link the tablet to the beacon you hit it with, overriding the previous beacon linked if there was one.")
+		linked_beacon = A
+		playsound(get_turf(src), 'sound/machines/triple_beep.ogg', 25, TRUE)
+		icon_state = "req_tablet_on"
+
+//Begin the requisitions copypasta
+/obj/item/req_tablet/interact(mob/user)
+	. = ..()
+	if(.)
+		return
+	var/dat
+	if(temp)
+		dat = temp
+	else
+		if (SSshuttle.supply)
+			if(!linked_beacon)
+				dat = "<BR><B>WARNING: NO CONNECTIVITY WITH SUPPLY SYSTEM DUE TO LACK OF LINKED BEACON!!</B>"
+			else
+				dat += "<BR><B>Automated Launch Pod Retrieval System: Tablet edition</B>"
+				dat += "<BR><HR>Supply points: [round(SSpoints.supply_points)]<BR>"
+				dat += "<BR><HR><A href='?src=\ref[src];order=categories'>Order supply packs (CANNOT BE CANCELED ONCE ORDERED)</A>"
+				dat += "<BR><HR><A href='?src=\ref[src];send=1'>Begin drop pod delivery</A>"
+				dat += "<BR><HR>WARNING: CAN ONLY LAUNCH ITEMS TO BEACON IF THE SURROUNDING TILES OF THE BEACON HAVE A GLASS OR NO CEILING PRESENT!!"
+				dat += "<BR><B>Current approved orders: </B><BR><BR>"
+				for(var/S in SSshuttle.shoppinglist)
+					var/datum/supply_order/SO = S
+					dat += "#[SO.id] - [SO.pack.name]<BR>"
+
+	var/datum/browser/popup = new(user, "computer", "<div align='center'>Ordering Tablet</div>", 575, 450)
+	popup.set_content(dat)
+	popup.open(FALSE)
+	onclose(user, "computer")
+
+/obj/item/req_tablet/Topic(href, href_list)
+	. = ..()
+	if(.)
+		return
+
+	if(href_list["order"])
```

Comment:
make sure to return within each `if(href_list["blah"])`

Reply:
If it returns then updateUsrDialog() wouldn't get called; as i don't want to put that proc in every single href list then return i didn't put in return for each href